dali/pipeline/data/tensor_list.h

```diff
@@ -154,8 +136,8 @@ class DLL_PUBLIC TensorList : private Buffer<Backend> {
     }
 
     use_copy_kernel &= (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned()) &&
-                       (std::is_same<Backend, GPUBackend>::value || pinned_);
-    type_.template Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
+                       (std::is_same<Backend, GPUBackend>::value || is_pinned());
+    type_info().template Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
```

Comment:
Nitpick: type_info is no longer a dependent name, so template disambiguation is not necessary (unless some buggy compiler thinks otherwise, of course).
```suggestion
    type_info().Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
```

Reply:
Not really, my compiler wasn't happy when I removed `template`.